src/classes/RD2_ValidationService.cls

```diff
@@ -196,6 +212,7 @@ public with sharing class RD2_ValidationService {
                 continue;
             }
 
+            validateRD2MetadataDeployed(rd);
```

Comment:
Do you think it's necessary to validate on a change too? I can see where this might catch RD's inserted between the time the org first enabled RD2 and when this change hits customer orgs, but (at least given the data we have now), it doesn't appears that like these customers have a lot of data if any. Mainly I'm thinking about it from a performance perspective.

Reply:
I'm agnostic about validating on change.  I suppose someone could get into a state where we needed to validate on update by doing a bulk load of inserts with triggers off, then re-enabling triggers for interactive access.

Reply:
@RadSFDeveloper What do you think?

Reply:
I'm less agnostic about this now.  If it is possible, given the current enablement flow, to get into a state where RDs can hit an update, we need to do this at least until the enablement flow is altered.  And we already know there have been records created, so we already know some records meet this test.  We need to validate on update.

Reply:
We can leave this for now, but maybe this gets added to the "refactor ideas" doc as a reminder that it should be pulled out at some point in the future once we get the enablement process under control.